server/controllers/logsController.ts

```diff
@@ -0,0 +1,155 @@
+  const retentionCutoffDate = () => {
+    // 10 days ago
+    return new Date(Date.now() - 10 * 24 * 60 * 60 * 1000);
+    // 2 seconds ago for testing purposes:
+    // return new Date(Date.now() - 2 * 1000);
+  };
+
+  // upload the cold log files from the local /logs directory to the database
+  const uploadLogs = async () => {
+    try {
+      // Read the logs directory
+      const files = await readdirAsync('./logs');
+
+      logger.debug('Uploading log files:', files);
+
+      // Process each log file
+      const logsUpdate: LogFile[] = files.map((filename) => {
+        const filePath = `./logs/${filename}`;
+        const fileData = readFileSync(filePath);
+        return { filename, fileData };
+      });
+
+      // Upload the logs to the database
+      await logsService.uploadLogs(logsUpdate);
+    } catch (error) {
+      logger.error('Error processing logs:', error);
+    }
+  };
+
+  const evictLogs = async () => {
+    // delete aged logs from the local directory
+    try {
+      // Read the logs directory
+      const files = await readdirAsync('./logs');
+
+      logger.debug('Evicting log files:', files);
+
+      for (const filename of files) {
+        // skip the hot log file
+        if (filename === 'linkta-server.log') return;
+        const filePath = `./logs/${filename}`;
+        // for Date created from yyMMdd pattern:
+        const fileCreatedAt = new Date(filename.split('.')[1]);
+        // next three lines for Date created from yyMMddhhmm pattern for testing purposes:
+        // const fileDateString = filename.split('.')[1]; // get date/time section from filename
+        // const dateString = `20${fileDateString.substring(0, 2)}-${fileDateString.substring(2, 4)}-${fileDateString.substring(4, 6)} ${fileDateString.substring(6, 8)}:${fileDateString.substring(8, 10)}:00`; // build a valid date/time string YYYY-MM-DD HH:MM:SS
+        // const fileCreatedAt = new Date(dateString); // create the date object for comparison
+        if (fileCreatedAt < retentionCutoffDate()) {
+          // delete the file
+          unlinkSync(filePath);
+        }
+      }
+    } catch (err) {
+      logger.error('Error reading log directory:', err);
+    }
+
+    // delete aged logs from the database
+    await logsService.evictLogs(retentionCutoffDate());
+  };
+
+  const getLogsByDateRange = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ) => {
+    logger.debug('Fetching logs by date range');
+
+    try {
+      const { startDate: startDateString, endDate: endDateString } = req.body;
+
+      const startDate = startDateString
+        ? new Date(startDateString)
+        : new Date(Date.now() - 30 * 24 * 60 * 60 * 1000); // 30 days ago
+
+      const endDate = endDateString ? new Date(endDateString) : new Date(); // Current date
+
+      // Validate dates
+      if (isNaN(startDate.getTime()) || isNaN(endDate.getTime())) {
+        throw new Error('Invalid startDate or endDate provided.');
```

Comment:
We can throw a 400 `BadRequestError` here instead for better UX

Reply:
Done.